third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -131,22 +146,31 @@ PjRtComputationClient::ExecuteComputation(
   const PjRtComputation& pjrt_computation =
       dynamic_cast<const PjRtComputation&>(computation);
 
+  xla::PjRtDevice* pjrt_device = StringToPjRtDevice(device);
+  XLA_CHECK(pjrt_device->IsAddressable()) << pjrt_device->DebugString();
+
   std::vector<xla::PjRtBuffer*> buffers;
   buffers.reserve(arguments.size());
   for (auto& argument : arguments) {
     const PjRtData* pjrt_data = dynamic_cast<PjRtData*>(argument.get());
+
+    XLA_CHECK(pjrt_device == pjrt_data->buffer->device())
+        << pjrt_device->DebugString() << " vs "
+        << pjrt_data->buffer->device()->DebugString();
     buffers.push_back(pjrt_data->buffer.get());
   }
 
   xla::ExecuteOptions execute_options;
-  execute_options.untuple_result = true;
-  std::vector<std::vector<std::unique_ptr<xla::PjRtBuffer>>> results =
-      pjrt_computation.executable->Execute({buffers}, execute_options)
+  execute_options.untuple_result = options.explode_tuple;
```

Comment:
What does this option do? 
 Maybe I am looking at a wrong struct: http://google3/third_party/tensorflow/compiler/xla/pjrt/pjrt_client.h;l=839;rcl=448855448

Could it be useful to leave a short comment, what & why we do this?

Reply:
This option corresponds 1:1 with the execute option we use in XRT to return a vector of tensors rather than a vector containing a tuple of tensors. We should probably document it there, though.

https://github.com/pytorch/xla/blob/1ab841877eaaa7a0acd6774172bcc52315f04073/third_party/xla_client/computation_client.h#L120-L122

Reply:
This option is explained here:

https://github.com/pytorch/xla/blob/1e7b2b25c5573e4e3dc06d54d02d9ae809ae79f2/third_party/xla_client/computation_client.h#L200-L201